https://inbox.sourceware.org/elfutils-devel/20230719231639.1786109-1-sam@gentoo.org/T/#u

From 7f607f9aba5c6fccb3d94776452b1cfa421df1f7 Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Thu, 20 Jul 2023 00:15:10 +0100
Subject: [PATCH] tests: run-lfs-symbols.sh needs gawk

With awk=mawk, I get:
```
FAIL: run-lfs-symbols.sh

First sanity-check that LFS detection works.
checking ./testfile-nolfs
awk: line 3: syntax error at or near /
FAIL run-lfs-symbols.sh (exit status: 2)
```

	* tests/run-lfs-symbols.sh: Call 'gawk' instead of 'awk'.

Signed-off-by: Sam James <sam@gentoo.org>
--- a/tests/run-lfs-symbols.sh
+++ b/tests/run-lfs-symbols.sh
@@ -42,13 +42,13 @@ LFS_FORMAT='BEGIN {
 LFS=$(printf "$LFS_FORMAT" "${abs_srcdir}/lfs-symbols")
 
 makeprint() {
-  make print-$1 -C $2 |& awk -F= "/^$1=/{ print \$2 }"
+  make print-$1 -C $2 |& gawk -F= "/^$1=/{ print \$2 }"
 }
 
 testrun_lfs() {
   echo "checking $1"
   if [ -e "$1" ]; then
-    bad=$(testrun ${abs_top_builddir}/src/nm -u "$1" | awk "$LFS")
+    bad=$(testrun ${abs_top_builddir}/src/nm -u "$1" | gawk "$LFS")
     if [ -n "$bad" ]; then
       echo "$1 contains non-lfs symbols:" $bad
       exit_status=1
-- 
2.41.0

